lib/players/computer.rb

```diff
@@ -0,0 +1,112 @@
+require_relative '../player.rb'
+
+class Player::Computer < Player
+
+	# Should make move based on Min Max Algorithm, where it searches results
+	# Should make move where all possible end results from this move results in the highest score
+	# Wins = +10, Loses = -10 and Draws = 0
+
+	attr_accessor :opponent, :choice
+
+	def initialize(token)
+		@token = token
+		if @token == "X"
+			@opponent = "O"
+		else
+			@opponent = "X"	
+		end
+	end
+
+	WIN_COMBINATIONS = [
+		[0,1,2], [3,4,5], [6,7,8],	# horizontal wins
+		[0,3,6], [1,4,7], [2,5,8],	# vertical wins
+		[0,4,8], [6,4,2]			# diagonal wins
+	]
+
+	def move(board)
+		new_board = board.cells.dup
+		minmax(new_board)
+		@choice
+	end
+
+
+	# Evaluates if the current player wins, loses or draws and sends back a score based on the result.
+	def score(board, depth)
+		if winner(board) == @token
+			return 10 - depth
+		elsif winner(board) == @opponent
+			return depth - 10
+		else
+			return 0
+		end
```

Comment:
implicit returns would work here so no need for `return`.

Also I think I'd use a case statement:

```ruby
case winner(board) ==
when @token
  10 - depth
when @opponent
  depth - 10
else
  0
end
```